tedana/io.py

```diff
@@ -436,28 +519,92 @@ def load_data(data, n_echos=None):
-        name.
+    ref_img : :obj:`str` or img_like
+        Reference image
+    data : (S [x T]) array_like
+        Data to be saved
+    affine : (4 x 4) array_like, optional
+        Transformation matrix to be used. Default: `ref_img.affine`
+    copy_header : :obj:`bool`, optional
+        Whether to copy header from `ref_img` to new image. Default: True
 
     Returns
     -------
-    comp_name : :obj:`str`
-        Component name in the form <prefix>_<zeros><comp_num>
+    nii : :obj:`nibabel.nifti1.Nifti1Image`
+        NiftiImage
     """
-    n_digits = int(np.log10(max_value)) + 1
-    comp_name = '{0:08d}'.format(int(comp_num))
-    comp_name = '{0}_{1}'.format(prefix, comp_name[8 - n_digits:])
-    return comp_name
+
+    ref_img = check_niimg(ref_img)
+    newdata = data.reshape(ref_img.shape[:3] + data.shape[1:])
+    if '.nii' not in ref_img.valid_exts:
+        # this is rather ugly and may lose some information...
+        nii = nib.Nifti1Image(newdata, affine=ref_img.affine,
+                              header=ref_img.header)
+    else:
+        # nilearn's `new_img_like` is a very nice function
+        nii = new_img_like(ref_img, newdata, affine=affine,
+                           copy_header=copy_header)
+    nii.set_data_dtype(data.dtype)
+
+    return nii
+
+
+def split_ts(data, mmix, mask, comptable):
+    """
+    Splits `data` time series into accepted component time series and remainder
+
+    Parameters
+    ----------
+    data : (S x T) array_like
+        Input data, where `S` is samples and `T` is time
+    mmix : (T x C) array_like
+        Mixing matrix for converting input data to component space, where `C`
+        is components and `T` is the same as in `data`
+    mask : (S,) array_like
+        Boolean mask array
+    comptable : (C x X) :obj:`pandas.DataFrame`
+        Component metric table. One row for each component, with a column for
+        each metric. Requires at least two columns: "component" and
+        "classification".
+
+    Returns
+    -------
+    hikts : (S x T) :obj:`numpy.ndarray`
+        Time series reconstructed using only components in `acc`
+    resid : (S x T) :obj:`numpy.ndarray`
+        Original data with `hikts` removed
+    """
+    acc = comptable[comptable.classification == 'accepted'].index.values
+
+    cbetas = get_coeffs(data - data.mean(axis=-1, keepdims=True),
+                        mmix, mask)
+    betas = cbetas[mask]
+    if len(acc) != 0:
+        hikts = utils.unmask(betas[:, acc].dot(mmix.T[acc, :]), mask)
+    else:
+        hikts = None
+
+    resid = data - hikts
+
+    return hikts, resid
+
+
+def check_convention() -> None:
+    """Checks set convention for io module
+
+    Raises
+    ------
+    RuntimeError, if invalid convention is set
+    """
+    if convention not in allowed_conventions:
+        raise RuntimeError(
+            ('Convention %s is not valid; allowed: ' % convention) +
+            str(allowed_conventions)
+        )
```

Comment:
```suggestion
        )
        

def set_convention(in_convention):
    """
    Parses user inputted convention to a subset of file naming conventions
    that are used in constants.py

    Parameters
    ----------
    in_convention : :obj:`str`
        A file naming convention inputted by the user
        Currently this is 'orig', 'bids' or a tedana version number, 
        like 0.0.9. If a version number is entered


    Returns
    -------
    convention : :obj:`str`
        A convention that is used in constants.py is globally set
        within this function
    """

    from packaging import version
    from .constants import bids, allowed_conventions

    if version.parse(in_convention).release:
        # If the convention is a version number
        # identify the correct convention for the tedana version
        version_num = version.parse(in_convention)
        if version_num <= version.parse("0.0.9"):
            convention = 'orig'
        else:
            convention = bids
    elif in_convention.casefold() == 'bids':
        # If the convention is 'bids' use the newest version
        # of the bids naming convention
        convention = bids
    else:
        # Otherwise, just assign the inputted convention and raise
        # error if the convention is not valid
        convention = in_convention
        check_convention()

    return convention

```